packages/main/src/sync.ts

```diff
@@ -25,23 +26,33 @@ const cachePolicy = {
 
 const BATCH_SIZE = 50
 
-export async function sync(accountId: cache.ID, manager: ConnectionManager) {
+async function boxSyncer(accountId: cache.ID, manager: ConnectionManager) {
+  let boxSync
+  let box
+
   for (const specifier of cachePolicy.boxes) {
-    const box = await manager
-      .request(request.actions.getBox(specifier))
-      .toPromise()
+    box = await manager.request(request.actions.getBox(specifier)).toPromise()
+
     const boxId = cache.persistBoxState(accountId, box)
-    await new BoxSync({
+
+    boxSync = await new BoxSync({
       accountId,
       box,
       boxId,
       manager
     }).sync()
   }
+  return boxSync
+}
+
+export async function sync(accountId: cache.ID, manager: ConnectionManager) {
   const contactApiClient = AccountManager.getContactsApiClient(
     String(accountId)
   )
-  contactApiClient && (await contactApiClient.downloadContacts(accountId))
+  const contactSync =
+    contactApiClient && (await contactApiClient.downloadContacts(accountId))
+
+  Promise.all([boxSyncer(accountId, manager), contactSync])
```

Comment:
This is good, but it is not parallel yet because you `await`ed on the `downloadContacts` call.